gateleen-routing/src/main/resources/gateleen_routing_schema_routing_rules

```diff
@@ -132,6 +136,32 @@
 			],
 			"additionalProperties": false
 		},
+		"ProxyOptions": {
+			"properties": {
+				"type": {
+                    "type": "string",
+                    "enum": [
+                        "HTTP",
+                        "SOCKS4",
+                        "SOCKS5"
+                    ]
+				},
+				"host": {
+					"type": "string"
```

Comment:
can we specify in this schema that this is a **required** attribute? (And also port - but not username/password)?

Reply:
AFAIK we could with:
```
"properties": { ... },
"additionalProperties": false,
"required": [ "host" , "port" ]
```

Reply:
of course we can. The question is whether we want or not? none of these properties are required since there's a default value (at least for type, host and port)

Reply:
I will add _host_ and _port_ as required since when someone wants to use the _proxyOptions_ property he should define something. I think the _type_ is not mandatory, because the default value is _HTTP_ which probably will be used in most cases.

Reply:
**host:** What default would you think for then? IMHO a default would be _no_ proxy. But then, whole _proxy_ object would make any sense.

**port:** Which defaults would be intuitive? IMHO that would depend on _type_. But even then it seems there are tons of different "defaults". Eg search for "proxy" on [List of TCP and UDP port numbers](https://en.wikipedia.org/wiki/List_of_TCP_and_UDP_port_numbers).